glustercli/cmd/tracing.go

```diff
@@ -0,0 +1,183 @@
+	Use:   "trace",
+	Short: helpTraceCmd,
+}
+
+var traceEnableCmd = &cobra.Command{
+	Use:   "enable <jaeger-endpoint> <jaeger-agent-endpoint>",
+	Short: helpTraceEnableCmd,
+	Args:  cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		jaegerEndpoint := args[0]
+		jaegerAgentEndpoint := args[1]
+
+		if jaegerEndpoint == "" || jaegerAgentEndpoint == "" {
+			err := errors.New("One or more Jaeger endpoints not specified")
+			failure(errTraceEnableReqFailed, err, 1)
+		}
+
+		if flagTraceJaegerSampler <= 0 || flagTraceJaegerSampler > 2 {
+			err := errors.New("Invalid Jaeger sampler specified")
+			failure(errTraceEnableReqFailed, err, 1)
+		}
+
+		if flagTraceJaegerSampler == 2 {
+			if flagTraceJaegerSampleFraction <= 0.0 || flagTraceJaegerSampleFraction >= 1.0 {
+				err := errors.New("Invalid Jaeger sample fraction specified")
+				failure(errTraceEnableReqFailed, err, 1)
+			}
+		}
+
+		_, err := client.TraceEnable(tracemgmtapi.SetupTracingReq{
+			JaegerEndpoint:       jaegerEndpoint,
+			JaegerAgentEndpoint:  jaegerAgentEndpoint,
+			JaegerSampler:        flagTraceJaegerSampler,
+			JaegerSampleFraction: flagTraceJaegerSampleFraction,
+		})
+
+		if err != nil {
+			if GlobalFlag.Verbose {
+				log.WithError(err).WithFields(log.Fields{
+					"JaegerEndpoint":       jaegerEndpoint,
+					"JaegerAgentEndpoint":  jaegerAgentEndpoint,
+					"JaegerSampler":        flagTraceJaegerSampler,
+					"JaegerSampleFraction": flagTraceJaegerSampleFraction,
+				}).Error("Trace Enable failed")
+			}
+			failure(errTraceEnableReqFailed, err, 1)
+		}
+		fmt.Println("Trace enable successful")
+	},
+}
+
+func samplerTypeToString(sampler int) string {
+	switch sampler {
+	case 0:
+		return "Never"
+	case 1:
+		return "Always"
+	case 2:
+		return "Probabilistic"
+	default:
+		return "Unknown"
+	}
+}
+
+var traceStatusCmd = &cobra.Command{
+	Use:   "status",
+	Short: helpTraceStatusCmd,
+	Run: func(cmd *cobra.Command, args []string) {
+		jaegerConfigInfo, err := client.TraceStatus()
+		if err != nil {
+			failure("Error getting trace status", err, 1)
+		}
+
+		table := tablewriter.NewWriter(os.Stdout)
+		table.SetAlignment(tablewriter.ALIGN_LEFT)
+		table.SetHeader([]string{"Trace Option", "Value"})
+
+		table.Append([]string{"Status", jaegerConfigInfo.Status})
+		table.Append([]string{"Jaeger Endpoint", jaegerConfigInfo.JaegerEndpoint})
+		table.Append([]string{"Jaeger Agent Endpoint", jaegerConfigInfo.JaegerAgentEndpoint})
+		table.Append([]string{"Jaeger Sampler", fmt.Sprintf("%d (%s)", jaegerConfigInfo.JaegerSampler, samplerTypeToString(jaegerConfigInfo.JaegerSampler))})
+		table.Append([]string{"Jaeger Sample Fraction", fmt.Sprintf("%0.2f", jaegerConfigInfo.JaegerSampleFraction)})
+		table.Render()
+		fmt.Println()
+	},
+}
+
+var traceUpdateCmd = &cobra.Command{
+	Use:   "update",
+	Short: helpTraceUpdateCmd,
+	Run: func(cmd *cobra.Command, args []string) {
+		if flagTraceJaegerSampler <= 0 || flagTraceJaegerSampler > 2 {
```

Comment:
I see the same validation in enable also, can this and below check be moved to a new function to avoid code duplication?

Reply:
Done. I have moved the common code to a function and used that.